### ISSUE TITLE
Remove deprecated t.python.threadpool.ThreadSafeList

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5473]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#5473|
|Type|task|
|Created|2012-02-05 13:30:29Z|
|Branch|https://github.com/twisted/twisted/tree/del-threadsafelist-5473|

`twisted.python.threadpool.ThreadSafeList` is deprecated in r28995 ([#3725](https://github.com/twisted/twisted/issues/3725)) and can be removed in a next release.

<details><summary>Searchable metadata</summary>

```
trac-id__5473 5473
type__task task
reporter__thijs thijs
priority__lowest lowest
milestone__ 
branch__branches_del_threadsafelist_5473 branches/del-threadsafelist-5473
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1328448629000000 1328448629000000
changetime__1328959778000000 1328959778000000
version__None None
owner__thijs thijs
cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

(In [33518]) Branching to 'del-threadsafelist-5473'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [33519]) remove ThreadSafeList and apply coding standard fixes. refs [#5473](https://github.com/twisted/twisted/issues/5473)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)|@jesstess set owner to @jesstess|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_5"></a>|@jesstess set owner to @thijstriemstra|
|-|-|

Thanks for working on this, thijs.

The [build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/del-threadsafelist-5473) show a failure in the documentation builder:

`twisted.python.threadpool.ThreadPool.callInThreadWithCallback: invalid ref to Deferred`

There are also a couple of places in `callInThreadWithCallback` which could use `C{}` markup to `onResult`.

Once those things are fixed, I think this is ready to merge!

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_6"></a>|@thijstriemstra commented|
|-|-|

(In [33536]) address review comments, refs [#5473](https://github.com/twisted/twisted/issues/5473)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `assigned`|
|-|-|

Replying to [jesstess](#note_5):
> Thanks for working on this, thijs.
> 
> The [build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/del-threadsafelist-5473) show a failure in the documentation builder:
> 
> `twisted.python.threadpool.ThreadPool.callInThreadWithCallback: invalid ref to Deferred`
> 
> There are also a couple of places in `callInThreadWithCallback` which could use `C{}` markup to `onResult`.
> 
> Once those things are fixed, I think this is ready to merge!

Thanks for the review. [Build results](http://buildbot.twistedmatrix.com/builders/documentation/builds/1841) are ok now, merging the branch.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_8"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [33537]) Merge del-threadsafelist-5473: Removed t.p.threadpool.ThreadSafeList (deprecated in 10.1), an internal implementation detail of support for Jython 2.1, which is now obsolete.

Author: thijs
Reviewer: jesstess
Fixes: [#5473](https://github.com/twisted/twisted/issues/5473)
